worlds/stick_ranger/Options.py

```diff
@@ -0,0 +1,439 @@
+
+    display_name = "Minimum number of required pre-Submarine Shrine stages"
+    range_start = 0
+    range_end = 12
+    default = 4
+
+
+class SubmarineShrineMaximumStagesUnlocks(Range):
+    """
+    Maximum number of pre-Submarine Shrine stages you must unlock before entering the Submarine Shrine stage.
+    Pre-Submarine Shrine stages are: Seaside 1-4, Submarine 1-4, Mist Grove 1-3 and ???.
+    """
+
+    display_name = "Maximum number of required pre-Submarine Shrine stages"
+    range_start = 0
+    range_end = 12
+    default = 7
+
+
+class StagesReqForSubmarineShrine(Range):
+    """
+    Actual number of required pre-Submarine Shrine stages to beat.
+    Only used for rules and tracking.
+    """
+
+    display_name = "Actual number of required pre-Submarine Shrine stages"
+    visibility = Visibility.none
+    range_start = 0
+    range_end = 12
+    default = 4
+
+
+class PyramidMinimumStagesUnlocks(Range):
+    """
+    Minimum number of pre-Pyramid stages you must unlock before entering the Pyramid stage.
+    Pre-Pyramid stages are: Desert 1-8, Oasis and Beach 1-3.
+    """
+
+    display_name = "Minimum number of required pre-Pyramid stages"
+    range_start = 0
+    range_end = 12
+    default = 4
+
+
+class PyramidMaximumStagesUnlocks(Range):
+    """
+    Maximum number of pre-Pyramid stages you must unlock before entering the Pyramid stage.
+    Pre-Pyramid stages are: Desert 1-8, Oasis and Beach 1-3.
+    """
+
+    display_name = "Maximum number of required pre-Pyramid stages"
+    range_start = 0
+    range_end = 12
+    default = 7
+
+
+class StagesReqForPyramid(Range):
+    """
+    Actual number of required pre-Pyramid stages to beat.
+    Only used for rules and tracking.
+    """
+
+    display_name = "Actual number of required pre-Pyramid stages"
+    visibility = Visibility.none
+    range_start = 0
+    range_end = 12
+    default = 4
+
+
+class IceCastleMinimumStagesUnlocks(Range):
+    """
+    Minimum number of pre-Ice Castle stages you must unlock before entering the Ice Castle stage.
+    Pre-Ice Castle stages are: Cavern 4-6, Snowfield 1-8, Mountain 1&2 and Frozen Lake.
+    """
+
+    display_name = "Minimum number of required pre-Ice Castle stages"
+    range_start = 0
+    range_end = 14
+    default = 5
+
+
+class IceCastleMaximumStagesUnlocks(Range):
+    """
+    Maximum number of pre-Ice Castle stages you must unlock before entering the Ice Castle stage.
+    Pre-Ice Castle stages are: Cavern 4-6, Snowfield 1-8, Mountain 1&2 and Frozen Lake.
+    """
+
+    display_name = "Maximum number of required pre-Ice Castle stages"
+    range_start = 0
+    range_end = 14
+    default = 8
+
+
+class StagesReqForIceCastle(Range):
+    """
+    Actual number of required pre-Ice Castle stages to beat.
+    Only used for rules and tracking.
+    """
+
+    display_name = "Actual number of required pre-Ice Castle stages"
+    visibility = Visibility.none
+    range_start = 0
+    range_end = 14
+    default = 5
```

Comment:
Should a player be allowed to change this option? If not, it probably shouldn't be an option, even if it's hidden. You can store information on the world class if you need to.